### DISCUSSION TITLE
New sidebar with menu

### DISCUSSION THREAD

**Discussion by suprunchuk** [shadcn-ui/ui#5480] [Q&A] [answered]
hi. I see that a new sidebar component has been introduced. but I want to make the top menu in the new sidebar like in the music template https://ui.shadcn.com/examples/music

1. I want to use a new sidebar https://ui.shadcn.com/blocks#sidebar-11
2. I want to always display the top menu at the top of the same page as in the music template https://ui.shadcn.com/examples/music

At the moment, the music template uses a custom sidebar. How can I use the top menu, which will be displayed everywhere and always, using the new sidebar?

I'm trying to add it to page.tsx, but it's blocked by the sidebar

![img-2024-10-21-10-45-19](https://github.com/user-attachments/assets/987ba24e-2c9f-4bcc-a353-9330c0a8ac62)

![img-2024-10-21-10-45-32](https://github.com/user-attachments/assets/c42df64c-d3dd-44d6-88c5-b30bb1c05878)

**Comment by enmanuelcanelon** [comment, marked as answer]
Hey @suprunchuk!

I ran into the same issue when the new sidebar was released. The trick is that you need to wrap your entire main content area (including your header) inside the `SidebarInset` component.

This component is specifically designed to handle the layout logic and padding so your content doesn't get "blocked" or overlapped by the sidebar.

Try structuring your page like this:

By putting the `<header>` inside the `SidebarInset`, it will respect the sidebar's width and transitions automatically, just like in the music template examples.Hope this helps!

```typescript
import { AppSidebar } from "@/components/app-sidebar"
import {
  SidebarInset,
  SidebarProvider,
  SidebarTrigger,
} from "@/components/ui/sidebar"
import { Separator } from "@/components/ui/separator"

export default function Page() {
  return (
    <SidebarProvider>
      <AppSidebar />
      <SidebarInset>
        {/* This header will now stay aligned with your content */}
        <header className="flex h-16 shrink-0 items-center gap-2 border-b px-4">
          <SidebarTrigger className="-ml-1" />
          <Separator orientation="vertical" className="mr-2 h-4" />
          <div className="flex items-center gap-2">
             <span className="font-semibold text-sm">Music App</span>
          </div>
        </header>
        
        {/* Your main content */}
        <main className="flex flex-1 flex-col gap-4 p-4">
          <div className="grid auto-rows-min gap-4 md:grid-cols-3">
            <div className="aspect-video rounded-xl bg-muted/50" />
            <div className="aspect-video rounded-xl bg-muted/50" />
            <div className="aspect-video rounded-xl bg-muted/50" />
          </div>
          <div className="min-h-[100vh] flex-1 rounded-xl bg-muted/50 md:min-h-min" />
        </main>
      </SidebarInset>
    </SidebarProvider>
  )
}
```
